File: test_merge_sort.f90
program test_merge_sort

use merge_sort_mod

implicit none

double precision :: a(3), b(3), c(6), d(7)
integer :: i

! demo merge 
a = (/ 1, 3, 5 /)
b = (/ 2, 4, 6 /)
call merge_arrays(a, 3, b, 3, c)
do i = 1, 6
  print '(a,i1,a,f9.4)', 'c(', i, ')=', c(i) 
end do


! demo merge sort
d = (/ 4, 3, 2, 4, 6, 7, 11 /)
call merge_sort(d, 7)
do i = 1, 7
  print '(a,i1,a,f9.4)', 'd(', i, ')=', d(i) 
end do

end program test_merge_sort

File: test_merge_sort.cc
#include <cstdio>

#include "merge.hh"
#include "merge_sort.hh"

int main()
{
  // demo merge
  {
    double a[3] = {1, 3, 5};
    double b[3] = {2, 4, 6};
    double c[6];
    merge(a, 3, b, 3, c);
    for (int i = 0; i < 6; ++i)
    {
      printf("c[%i]=%.2f\n", i, c[i]); 
    }
  }

  // demo merge sort
  {
    int n = 7;
    double a[n] = {4, 3, 2, 4, 6, 7, 11};
    merge_sort(a, n);
    for (int i = 0; i < n; ++i)
    {
      printf("a[%i]=%.2f\n", i, a[i]); 
    }
  }

  return 0;
}
